src/examples/common_plane_gpu.cpp

```diff
@@ -0,0 +1,280 @@
+
+    if (mem_space == tribol::MemorySpace::Device)
+    {
+      err = runExample<tribol::MemorySpace::Device, tribol::ExecutionMode::Hip>(mesh_file, num_contact_elems);
+    }
+    else
+#endif
+    {
+#ifdef TRIBOL_USE_OPENMP
+      auto exec_mode = tribol::ExecutionMode::Sequential;
+      if (exec_mode == tribol::ExecutionMode::OpenMP)
+      {
+        err = runExample<tribol::MemorySpace::Host, tribol::ExecutionMode::OpenMP>(mesh_file, num_contact_elems);
+      }
+      else
+#endif
+      {
+        err = runExample<tribol::MemorySpace::Host, tribol::ExecutionMode::Sequential>(mesh_file, num_contact_elems);
+      }
+    }
+  }
+
+  axom::slic::flushStreams();
+  finalize_logger();
+
+#ifdef TRIBOL_USE_MPI
+  MPI_Finalize();
+#endif
+
+  return err;
+}
+
+template <tribol::MemorySpace MSPACE, tribol::ExecutionMode EXEC>
+int runExample(const std::string& mesh_file, int num_contact_elems)
+{
+
+  mfem::Device device;
+  switch (MSPACE)
+  {
+#ifdef TRIBOL_USE_CUDA
+    case tribol::MemorySpace::Device:
+      device.Configure("cuda");
+      break;
+#endif
+    default:
+// #ifdef TRIBOL_USE_OPENMP
+//       if (EXEC == tribol::ExecutionMode::OpenMP)
+//       {
+//         device.Configure("omp");
+//       }
+//       else
+// #endif
+      {
+        device.Configure("cpu");
+      }
+      break;
+  }
+  device.Print();
+
+  std::cout << "Creating MFEM mesh..." << std::endl;
+  axom::utilities::Timer timer(true);
+
+  mfem::Mesh mesh(mesh_file);
+
+  int ref_ct = 0;
+  for (int i{0}; i < ref_ct; ++i)
+  {
+    mesh.UniformRefinement();
+  }
+  std::cout << "MFEM mesh created (" << timer.elapsedTimeInMilliSec() << " ms)" << std::endl;
+
+  std::cout << "Creating MFEM grid function..." << std::endl;
+  timer.start();
+  mfem::H1_FECollection fe_coll(1, mesh.SpaceDimension());
+  mfem::FiniteElementSpace fe_space(&mesh, &fe_coll, mesh.SpaceDimension());
+  mfem::GridFunction coords(&fe_space);
+  mesh.SetNodalGridFunction(&coords, false);
+
+  auto coords_ptr = coords.Read();
+  auto x_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 0)];
+  auto y_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 1)];
+  auto z_coords_ptr = &coords_ptr[fe_space.DofToVDof(0, 2)];
+  std::cout << "MFEM grid function created (" << timer.elapsedTimeInMilliSec() << " ms)" << std::endl;
+
+  std::cout << "Creating Tribol connectivity..." << std::endl;
+  timer.start();
+
+  // mesh 1 connectivity (build on cpu)
+  auto mesh1_bdry_attrib = 4;
+  tribol::ArrayT<tribol::IndexT, 2, tribol::MemorySpace::Host> host_conn_1(num_contact_elems, 4);
+  int elem_ct = 0;
+  for (int be{0}; be < mesh.GetNBE(); ++be)
+  {
+    if (mesh.GetBdrAttribute(be) == mesh1_bdry_attrib)
+    {
+      mfem::Array<int> be_dofs(4);//, mfem::MemoryType::Host_UMPIRE);
+      fe_space.GetBdrElementDofs(be, be_dofs);
+      for (int i{0}; i < 4; ++i)
+      {
+        host_conn_1(elem_ct, i) = be_dofs[i];
+      }
+      ++elem_ct;
+    }
+  }
+  // move to gpu
```

Comment:
Maybe clarify that won't this only move to gpu only for certain memory spaces? That is, it may not be obvious that creating an ArrayT will "move" data to the gpu.